src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -315,6 +315,38 @@ public boolean validateParticipant(DocumentReference regDocRef, String emailAdr,
     return success;
   }
 
+  @Override
+  public CourseConfirmState getConfirmState(DocumentReference regDocRef) {
+    DocumentReference courseParticipantClassRef = getCourseClasses().getCourseParticipantClassRef(
+        modelContext.getWikiRef().getName());
+    CourseConfirmState confirmState = CourseConfirmState.UNDEFINED;
+    try {
+      for (BaseObject obj : modelAccess.getXObjects(regDocRef, courseParticipantClassRef)) {
+        Optional<CourseConfirmState> state = CourseConfirmState.convertStringToEnum(
+            obj.getStringValue("status"));
+        if (state.isPresent()) {
+          switch (confirmState) {
```

Comment:
i think the switch and break statements make this part hard to understand. wouldn't this be much easier:
```java
if (confirmState == CourseConfirmState.UNDEFINED) {
  confirmState = state.get();
} else if (confirmState != state.get()) {
  confirmState = CourseConfirmState.PARTIALCONFIRMED;
  break;
}
```